api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -676,6 +676,12 @@ message Cluster {
     // If this is unset then [UNKNOWN, HEALTHY, DEGRADED] will be applied by default. If this is
     // set with an empty set of statuses then host overrides will be ignored by the load balancing.
     core.v3.HealthStatusSet override_host_status = 8;
+
+    // This controls what policy is used to extract the override host from the downstream context
```

Comment:
drive-by comment:
Is the goal of this PR to define a *per-cluster* override host mechanism (compared to the per-route mechanism, e.g., using the affinity-filter that is defined for a route)?
If so, I wonder if a better mechanism that allows declaring per-cluster filters, and we can formally reconcile with per-route filters will work better in the long run. In other words, it could be that the per-cluster override-host-policy is an instance of the per-cluster filter?

Reply:
Partially yeah. Another point is we want to move all these related feature to upstream/cluster's configuration.

We have supported per-cluster fiters actually, the upstream http filters, but it works after host selection, so cannot be used in this scenario.